instagram-app/src/authentication/withAuthenticate.js

```diff
@@ -0,0 +1,32 @@
+import React from 'react'
+
+const withAuthenticate = App => Login =>
+    class extends React.Component {
+        state = {
+            loggedIn: false
+        }
+
+    componentDidMount() {
+        if(localStorage.getItem("username")) {
+            this.setState({
+                loggedIn: true
+            })
+        }
+        else {
+            this.setState({
+                loggedIn: false
+            })
+        }
+    }
+
+    render() {
+        if(this.state.loggedIn) {
+        return <App />
+        } else {
```

Comment:
To make this a bit cleaner you could use a [Ternary Operator](https://developer.mozilla.org/enUS/docs/Web/JavaScript/Reference/Operators/Conditional_Operator) here. To do that it would like something like this `this.state.loggedIn ? <App/> : <Login/>`